data/src/main/kotlin/com/alfresco/content/data/SearchRepository.kt

```diff
@@ -212,15 +216,68 @@ class SearchRepository {
             null
         }
 
-    suspend fun getRecents(skipCount: Int, maxItems: Int) =
-        ResponsePaging.with(
-            searchService.recentFiles(
-                session.account.id,
-                MAX_RECENT_FILES_AGE,
-                skipCount,
-                maxItems,
-            ),
-        )
+    suspend fun getRecents(skipCount: Int, maxItems: Int): ResponsePaging {
+        val version = getServerVersion()
+
+        if (version.toInt() >= 23) {
+            return ResponsePaging.with(
+                searchService.recentFiles(
+                    session.account.id,
+                    MAX_RECENT_FILES_AGE,
+                    skipCount,
+                    maxItems,
+                    true,
+                ),
+            )
+        } else {
+            return ResponsePaging.with(
+                searchService.recentFiles(
+                    session.account.id,
+                    MAX_RECENT_FILES_AGE,
+                    skipCount,
+                    maxItems,
+                ),
+            )
+        }
+    }
+
+    fun getPrefsServerVersion(): String {
+        val sharedPrefs = PreferenceManager.getDefaultSharedPreferences(context)
+        return sharedPrefs.getString(SERVER_VERSION, "") ?: ""
+    }
+
+    private fun saveServerVersion(serverVersion: String) {
+        val sharedPrefs = PreferenceManager.getDefaultSharedPreferences(context)
+        val editor = sharedPrefs.edit()
+        editor.putString(SERVER_VERSION, serverVersion)
+        editor.apply()
```

Comment:
On logout, this saved data needs to be clear.

Reply:
done